legend-engine-core-executionPlan-execution/legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/graphFetch/ParallelGraphFetchExecutionConfig.java

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.plan.execution.graphFetch;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class ParallelGraphFetchExecutionConfig
+{
+    public final long DEFAULT_DEFAULT_MAX_CONNECTIONS_PER_DATABASE = 10;
+    public final long DEFAULT_PARALLEL_EXECUTION_POOL_SIZE = 100;
+
+    @JsonProperty
+    public final long parallelExecutionPoolSize;
+    @JsonProperty
+    private final long defaultMaxConnectionsPerDatabase;
+    @JsonProperty
+    private final Map<String, Integer> maxConnectionsPerDatabaseTypeOverrides;
```

Comment:
Database concepts cannot come into this space. You should introduce a relational extension concept

Reply:
moved relational configuration into https://github.com/finos/legend-engine/pull/2077/files#diff-e8404bcc8de985e707449918d183939695512ba0fdcab75c1e90089a2ab48d29. 
pool is also store agnostic: https://github.com/finos/legend-engine/pull/2077/files#diff-48aee8e3116e7f91188d4476f182eb43ba061092ff6d86cd53f14adfeb0a19b6.